### ISSUE TITLE
Unified BitVector Tests

### ISSUE THREAD

**Issue by cdonovick** [leonardt/hwtypes#126]
There have been and continue to be a number of areas of where `[SMT]BitVector` and `Bits` have diverged on the typing of methods and operators.  I propose building a unified test sweet which tests the typing of operators that both magma and hwtypes run.  

Now I don't want to require that an implementation necessarily have constant propagation or be castable to int so we can't test the values produced.  However, the type relationships should be consistent.  

Basically I want to make https://github.com/leonardt/hwtypes/blob/master/tests/test_optypes.py more complete and have magma run it as well.
